paparazzi/src/test/java/app/cash/paparazzi/HtmlReportWriterTest.kt

```diff
@@ -266,9 +267,106 @@ class HtmlReportWriterTest {
+      val convertedVertical = vertical.convertImage()
+      val square = ImageIO.read(File("src/test/resources/square.png"))
+      val convertedSquare = square.convertImage()
+      val htmlReportWriter = HtmlReportWriter("record_run", reportRoot.root, snapshotRoot.root)
+
+      htmlReportWriter.use {
+        val horizontalFrameHandler = htmlReportWriter.newFrameHandler(
+          Snapshot(
+            name = "horizontal",
+            testName = TestName("app.cash.paparazzi", "SimilarViews", "horizontal"),
+            timestamp = Instant.parse("2019-03-20T10:27:43Z").toDate(),
+            tags = listOf("redesign")
+          ),
+          frameCount = 1,
+          fps = -1
+        )
+        horizontalFrameHandler.use {
+          horizontalFrameHandler.handle(convertedHorizontal)
+        }
+        val verticalFrameHandler = htmlReportWriter.newFrameHandler(
+          Snapshot(
+            name = "loading",
+            testName = TestName("app.cash.paparazzi", "SimilarViews", "vertical"),
+            timestamp = Instant.parse("2019-03-20T10:27:43Z").toDate(),
+            tags = listOf("redesign")
+          ),
+          frameCount = 1,
+          fps = -1
+        )
+        verticalFrameHandler.use {
+          verticalFrameHandler.handle(convertedVertical)
+        }
+        val squareFrameHandler = htmlReportWriter.newFrameHandler(
+          Snapshot(
+            name = "loading",
+            testName = TestName("app.cash.paparazzi", "SimilarViews", "square"),
+            timestamp = Instant.parse("2019-03-20T10:27:43Z").toDate(),
+            tags = listOf("redesign")
+          ),
+          frameCount = 1,
+          fps = -1
+        )
+        squareFrameHandler.use {
+          squareFrameHandler.handle(convertedSquare)
+        }
+      }
+      // Verify that there are 3 files in the reports/paparazzi/images file
+      // This should confirm that the snapshot images used in the test report are correct
+      val snapshotDir = File(snapshotRoot.root, "images").listFiles()!!
+      assertThat(snapshotDir.size).isEqualTo(3)
+
+      val firstSnapshot = ImageIO.read(snapshotDir[0])
+      val secondSnapshot = ImageIO.read(snapshotDir[1])
+      val thirdSnapshot = ImageIO.read(snapshotDir[2])
+      assertThat(firstSnapshot.pixelEqual(secondSnapshot)).isFalse()
+      assertThat(firstSnapshot.pixelEqual(thirdSnapshot)).isFalse()
+      assertThat(secondSnapshot.pixelEqual(thirdSnapshot)).isFalse()
+    } finally {
+      // reset record mode
+      System.setProperty("paparazzi.test.record", "false")
+    }
+  }
+
   private fun Instant.toDate() = Date(toEpochMilli())
 
   private fun File.lastModifiedTime(): FileTime {
     return Files.readAttributes(this.toPath(), BasicFileAttributes::class.java).lastModifiedTime()
   }
+
+  // Function that takes in an existing Buffered image and converts its type to one
+  // that ApngWriter can consume
+  private fun BufferedImage.convertImage() = BufferedImage(
```

Comment:
This is needed because the test images do not have a color type that works with `ApngWriter`. I find it odd that I need to do this since I took the images from a test project that the `ApngWriter` created. Do we need to expand the list of valid image types for the writer?